### PR TITLE
build/cy4390x : Add defconfig and make file for st_things.

### PR DESCRIPTION
**Pull request by uvasistha-samsung**
Add defconfig and make file to build TizenRT and flash binary for st_things module for cy43909.

Signed-off-by: uvasistha-samsung <utkarsh.m@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b6a888d4e9ec951eb957bf7d8c79245ad09f9099] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b6a888d4e9ec951eb957bf7d8c79245ad09f9099] - Code Rule Check (C++) OK.

**Comment by SenthilKumarGS** [comment]
@uvasistha-samsung 
I guess we will have to move these files under 'build/configs/cy4390x'.

**Comment by SenthilKumarGS** [comment]
@uvasistha-samsung 
I guess we will have to move these files under 'build/configs/cy4390x'.

File: build/configs/st_things/Make.defs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ARCHWARNINGS = -Wall -Wstrict-prototypes -Wshadow -Wundef
+ARCHWARNINGSXX = -Wall -Wshadow -Wundef
+ifeq ($(CONFIG_ARM_UNWIND),y)
+  ARCHDEFINES = -fasynchronous-unwind-tables -mno-sched-prolog
+endif
+ARCHPICFLAGS = -fpic -msingle-pic-base -mpic-register=r10
+
+CFLAGS = $(ARCHCFLAGS) $(ARCHWARNINGS) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe
+CPICFLAGS = $(ARCHPICFLAGS) $(CFLAGS)
+CXXFLAGS = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHXXINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe
```
Comment (sunghan-chang) [inline build/configs/st_things/Make.defs]: Could you add below at next this line?
```
ifeq ($(CONFIG_LIBCXX),y)
CXXFLAGS += -DCONFIG_WCHAR_BUILTIN
endif
```

All of latest Make.defs has it.

**Comment by sunghan-chang** [comment]
@uvasistha-samsung As @SenthilKumarGS said, you must put it in build/configs/cy4390x.
build/configs/cy4390x/st_things/defconfigs and Make.defs

File: build/configs/st_things/Make.defs (hunk 1 above)
Comment (uvasistha-samsung) [reply build/configs/st_things/Make.defs]: Okay Adding these changes

File: build/configs/st_things/Make.defs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+endif
+
+HOSTCC = gcc
+HOSTINCLUDES = -I.
+HOSTCFLAGS = -Wall -Wstrict-prototypes -Wshadow -Wundef -g -pipe
+HOSTLDFLAGS =
+
+define DOWNLOAD
+  @$(TOPDIR)/../build/configs/cy4390x/cy4390x_download.sh $(1)
+endef
```
[the remark this answers is not in the document]
Comment (uvasistha-samsung) [reply build/configs/st_things/Make.defs]: Okay

**Comment by seinfra** [comment]
Target : [86dec1921bf48d8414e7111c5251241046a4f56a] - Code Rule Check Result:
apps/examples/mediaplayer/BufferInputDataSource.h:25: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
apps/examples/mediaplayer/BufferInputDataSource.h:25: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
apps/examples/mediaplayer/BufferInputDataSource.h:25: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
apps/examples/mediaplayer/BufferInputDataSource.h:25: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
apps/examples/mediaplayer/BufferInputDataSource.h:25: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
apps/examples/mediaplayer/BufferInputDataSource.h:39: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
apps/examples/mediaplayer/BufferInputDataSource.h:39: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
apps/examples/mediaplayer/BufferInputDataSource.h:39: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
apps/examples/testcase/ta_tc/media/utc/gmock/utc_media_common.h:29: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
apps/examples/testcase/ta_tc/media/utc/gmock/utc_media_common.h:31: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxO)
apps/examples/testcase/ta_tc/media/utc/gmock/utc_media_common.h:31: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
apps/examples/testcase/ta_tc/media/utc/gmock/utc_media_common.h:31: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
apps/examples/testcase/ta_tc/media/utc/gmock/utc_media_common.h:31: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by sunghan-chang** [comment]
@uvasistha-samsung Here is no update. Could I close this?

**Comment by uvasistha-samsung** [comment]
Due to some git issues we couldn't update this PR .Hence created a new PR 
(https://github.com/Samsung/TizenRT/pull/2473)
Please Follow the link above ...

**Pull request closed without merging**